src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -257,6 +259,7 @@ private void ClearCompilationScopeMap(AnalyzerAndOptions analyzerAndOptions, Ana
             Tuple<ImmutableArray<DiagnosticDescriptor>, EventHandler<Exception>> value;
             lock (_descriptorMap)
             {
+                // REVIEW: how one knows exception handler is same as the one cached?
```

Comment:
Oh i see.  Yeah, tha'ts definitely concerning.  But probably not that big a deal in practice (i hope).